### ISSUE TITLE
StockItemCriteria::setProductsFilter doesn't work with array of ids

### ISSUE THREAD

**Issue by mcjwsk** [magento/magento2#7678]
### Preconditions
1. Magento 2.1.2 EE
2. PHP 7.0.9 

### Steps to reproduce
```php
/** @var \Magento\CatalogInventory\Api\StockItemCriteriaInterface $criteria */
$criteria = $this->criteriaFactory->create();
$criteria->setProductsFilter([1, 3, 5, 7, 9]);
$collection = $this->stockItemRepository->getList($criteria);
echo $collection->getTotalCount();
```

### Expected result
```
5
```

### Actual result
```
1
```

It works if you change the product filter to:
```php
/** @var \Magento\CatalogInventory\Api\StockItemCriteriaInterface $criteria */
$criteria = $this->criteriaFactory->create();
$criteria->setProductsFilter([[1, 3, 5, 7, 9]]);
$collection = $this->stockItemRepository->getList($criteria);
echo $collection->getTotalCount(); // 5
```

This issue is a duplicate of the ticket #4328

**Comment by AydinHassan** [comment]
I have also just experienced this today - can confirm a bug.

**Comment by elvin-vaimo** [comment]
See #4328 , there is also workaround.

**Comment by magento-engcom-team** [comment]
@DariuszMaciejewski, thank you for your report.
We've created internal ticket(s) MAGETWO-81245 to track progress on the issue.

**Comment by kirmorozov** [comment]
Will take care of it.  #SQUASHTOBERFEST

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82004

**Comment by okorshenko** [comment]
the issue has been fixed in 2.2-develop branch and will be available in 2.2.2 release
